src/hooks/useEditorView.ts

```diff
@@ -1,56 +1,33 @@
-import type { EditorState, Transaction } from "prosemirror-state";
+import { Schema } from "prosemirror-model";
+import { EditorState } from "prosemirror-state";
+import type { Plugin, Transaction } from "prosemirror-state";
 import { EditorView } from "prosemirror-view";
-import type { DirectEditorProps } from "prosemirror-view";
-import { useLayoutEffect, useState } from "react";
+import type { EditorProps } from "prosemirror-view";
+import { useLayoutEffect, useMemo, useState } from "react";
 import { flushSync } from "react-dom";
 
-import { useForceUpdate } from "./useForceUpdate.js";
+import type { EditorContextValue } from "../contexts/EditorContext.js";
 
-function withFlushedUpdates<This, T extends unknown[]>(
-  fn: (this: This, ...args: T) => void
-): (...args: T) => void {
-  return function (this: This, ...args: T) {
-    flushSync(() => {
-      fn.call(this, ...args);
-    });
-  };
-}
```

Comment:
I think this file is easier to read now without this higher-order function making things more abstract.